### PR TITLE
Update/modernize CONTRIBUTING guide

### PR DESCRIPTION
**Pull request by MichaelChirico**
Follow-up to #5859, again OTW to #5847.

One point that's unclear to me -- we refer several times to "project members" as anyone that has the permissions to create branches.

Do we have any distinction between people that can create branches and those that can merge branches? If so what terminology should we use respectively?

### CODE REVIEW

**Comment by tdhock** [comment]
In governance, committer is the role which can merge PR into master, https://github.com/Rdatatable/data.table/blob/master/GOVERNANCE.md#committer

**Review by tdhock — APPROVED** [review]
looks good

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -34,67 +41,69 @@ A few minor points of style that you should adhere to in your PR:
...
-See [`test` function manual](https://rdatatable.gitlab.io/data.table/reference/test.html).
+See [`?test`](https://rdatatable.gitlab.io/data.table/reference/test.html).
 
 **References:** If you are not sure how to issue a PR, but would like to contribute, these links should help get you started:
 
 1. **[How to Github: Fork, Branch, Track, Squash and Pull request](https://gun.io/blog/how-to-github-fork-branch-and-pull-request/)**.
-2. **[Squashing Github pull requests into a single commit](http://eli.thegreenplace.net/2014/02/19/squashing-github-pull-requests-into-a-single-commit)**.
-3. **[Github help](https://help.github.com/articles/using-pull-requests/)** - you'll need the *fork and pull* model.
+1. **[Squashing Github pull requests into a single commit](http://eli.thegreenplace.net/2014/02/19/squashing-github-pull-requests-into-a-single-commit)**.
+1. **[Github help](https://help.github.com/articles/using-pull-requests/)** - you'll need the *fork and pull* model.
```
Comment (jangorecki) [inline .github/CONTRIBUTING.md:63]: We have triple 1. in line prefixes

Comment (MichaelChirico) [reply .github/CONTRIBUTING.md:63]: yes that's intentional. note the rendered version:

Screenshot_2023-12-30-15-47-46-32_40deb401b9ffe8e1df2f1cc5ba480b12

using repeated 1. lets the list be rearranged later without a huge distracting diff

**Pull request merged**